### ISSUE TITLE
Only renders on bottom left in Music app (macOS)

### ISSUE THREAD

**Issue by jeffijoe** [projectM-visualizer/projectm#332]
**OS**: macOS Catalina
**ProjectM Version**: 3.1.3
**MacBook**: 2019 model, 16''

![image](https://user-images.githubusercontent.com/2505474/76626082-df635d00-650e-11ea-9547-f81a7ea609ae.png)

Thank you for keeping MilkDrop alive!

**Comment by rullinoiz** [comment]
how did you get the plugin to work with music? also i've heard running the app in low resolution mode (app > right click > get info > run in low resolution mode) gets it back to full screen

**Comment by jeffijoe** [comment]
@rullinoiz just tried, that didn't fix it.

I just installed the plugin, opened Music, then did this:

![image](https://user-images.githubusercontent.com/2505474/78461899-dfccbf00-769a-11ea-803f-30b80ba4f307.png)

**Comment by dreness** [comment]
I got the same behavior, which looks a lot like one of the common ways things can go wrong on high-dpi displays. So, I poked around to find where the frame size is set and just asked for twice as much in both dimensions:

```diff
❯ git diff src/projectM-iTunes/iprojectM_mac.mm
diff --git a/src/projectM-iTunes/iprojectM_mac.mm b/src/projectM-iTunes/iprojectM_mac.mm
index bb7b6a11..4ed135b7 100644
--- a/src/projectM-iTunes/iprojectM_mac.mm
+++ b/src/projectM-iTunes/iprojectM_mac.mm
@@ -228,8 +228,8 @@ OSStatus ResizeVisual( VisualPluginData * visualPluginData )
        visualPluginData->destRect = [visualPluginData->subview bounds];
 
     if (visualPluginData->pm != NULL) {
-        visualPluginData->pm->projectM_resetGL(visualPluginData->destRect.size.width, visualPluginData->destRect.size.height);
-        NSLog(@"resized to %@ %@", [NSNumber numberWithDouble: visualPluginData->destRect.size.width], [NSNumber numberWithDouble: visualPluginData->destRect.size.height]);
+        visualPluginData->pm->projectM_resetGL(visualPluginData->destRect.size.width*2, visualPluginData->destRect.size.height*2);
+        NSLog(@"resized to %@ %@", [NSNumber numberWithDouble: visualPluginData->destRect.size.width*2], [NSNumber numberWithDouble: visualPluginData->destRect.size.height*2]);
         
         visualPluginData->readyToDraw = true;
     }

```


It's hard to see in the diff, but basically `width` becomes `width*2`, `height` becomes `height*2`. With that change, things are back to occupying the entire window:

<img width="1142" alt="Screen Shot 2020-04-17 at 3 55 48 AM" src="https://user-images.githubusercontent.com/5242016/79564124-e5df7a00-8062-11ea-96c3-5a6a8740451e.png">

Of course this is a dumb hack, but serves to illustrate the cause of the problem, I believe.

While I'm here: here's a helpful log query - note the use of `--last 20m` to limit the output to the most recent 20 minutes of logs.
```
❯ log show --info --debug --predicate "(process = 'VisualizerService') and (senderImagePath endsWith 'iProjectM')" --last 20m --style compact
Filtering the log data using "process == "VisualizerService" AND senderImagePath ENDSWITH "iProjectM""
Timestamp               Ty Process[PID:TID]
2020-04-17 04:11:36.769 Df VisualizerService[28129:e2093b] (iProjectM) initWithFrame called
2020-04-17 04:11:36.829 Df VisualizerService[28129:e2093b] (iProjectM) super initWithFrame called
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) activate visual
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) main bundle: /Users/andre/Library/iTunes/iTunes Plug-ins/iProjectM.bundle
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) presets path /Users/andre/Library/iTunes/iTunes Plug-ins/iProjectM.bundle/Contents/Resources/presets
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) GL_VERSION: 4.1 INTEL-14.5.22
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) GL_SHADING_LANGUAGE_VERSION: 4.10
2020-04-17 04:11:36.832 Df VisualizerService[28129:e2093b] (iProjectM) GL_VENDOR: Intel Inc.
2020-04-17 04:11:37.618 Df VisualizerService[28129:e2093b] (iProjectM) random selected
2020-04-17 04:11:37.729 Df VisualizerService[28129:e2093b] (iProjectM) resized to 960 540
2020-04-17 04:11:37.729 Df VisualizerService[28129:e2093b] (iProjectM) activated visual
2020-04-17 04:11:37.840 Df VisualizerService[28129:e2093b] (iProjectM) resized to 960 540
2020-04-17 04:11:38.140 Df VisualizerService[28129:e2093b] (iProjectM) resized to 960 540
2020-04-17 04:11:38.287 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2060 1492
2020-04-17 04:13:01.996 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2060 1540
2020-04-17 04:13:02.107 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2160 3840
2020-04-17 04:13:02.223 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2160 2096
2020-04-17 04:13:02.334 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2060 1540
2020-04-17 04:13:02.603 Df VisualizerService[28129:e2093b] (iProjectM) resized to 3584 2240
2020-04-17 04:13:39.669 Df VisualizerService[28129:e2093b] (iProjectM) resized to 3584 2192
2020-04-17 04:13:39.782 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2160 3792
2020-04-17 04:13:39.893 Df VisualizerService[28129:e2093b] (iProjectM) resized to 2060 1492
```

**Comment by ghodss** [comment]
I am seeing the same behavior as described in this issue.

**Comment by ttakumat** [comment]
same problem here

**Comment by revmischa** [comment]
Should be fixed in 44ab74e
